### PR TITLE
[1.10] Admin Router: Remove authz from /metadata endpoint

### PR DESCRIPTION
**Pull request by gpaul**
## High Level Description

This is a backport of https://github.com/dcos/dcos/commit/277a61651e74f06a5c80105421a6e7c7a9d398b3 (originally in https://github.com/mesosphere/dcos/pull/43) to the 1.10 branch.

This PR removes authz from `/metadata` endpoint. Please see linked Jira issues for the reasoning behind this PR and other details.

## Related Issues

  - [DCOS-14563](https://jira.mesosphere.com/browse/DCOS-14563) Admin Router: switch /metadata to "allow all authenticated users" instead of ACL lookup

## Checklist for all PR's

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here: Test added downstream in enterprise.
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and Review Process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using github's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by gpaul** [comment]
@vespian @mhrabovcin PTAL

**Comment by gpaul** [comment]
@mesosphere-mergebot bump-ee

**Comment by gpaul** [comment]
@mesosphere-mergebot label Ready For Review

File: packages/adminrouter/extra/src/lib/auth/open.lua [hunk 1]
```diff
@@ -108,9 +108,13 @@ function _M.init(use_auth)
         return res.do_authn_and_authz_or_exit()
     end
 
-    -- /metadata
     -- /pkgpanda/active.buildinfo.full.json
     -- /dcos-metadata/
+    res.access_misc_metadata_endpoint = function()
+        return res.do_authn_and_authz_or_exit()
```
Comment (mhrabovcin) [inline packages/adminrouter/extra/src/lib/auth/open.lua]: This function call hints it does the `authz`.

Comment (vespian) [reply packages/adminrouter/extra/src/lib/auth/open.lua]: In open authn is also authz.

https://github.com/dcos/dcos/blob/3847b7e21047994c84cac58aabed09d39b9206fd/packages/adminrouter/extra/src/lib/auth/open.lua#L42-L43

**Comment by gpaul** [comment]
@vespian this is a clean backport (read: cherry-pick) of a 1.11 PR - if there are missing tests the original never had any. If there are commits that apply cleanly to the 1.10 branch that add tests we can backport them, too. Otherwise I'd say the backport is not the place to do new work.

**Comment by gpaul** [comment]
@mhrabovcin I've basically zero context for this backport PR other than it is a clean cherry-pick of the single commit contained in https://github.com/mesosphere/dcos/pull/43. 

We do leak mention of authz into the dcos/dcos repo, yes: https://github.com/dcos/dcos/blob/c1ae4514b07ece64613a833b61f6ac38a259d4c3/packages/adminrouter/extra/src/lib/auth/open.lua#L39

**Comment by gpaul** [comment]
@vespian Ah! Makes sense, thanks for clarifying. I'm totally out of my depth with this backport :D

**Comment by gpaul** [comment]
@mesosphere-mergebot ship_it_label_status_check

**Pull request merged**